pkg/operation/botanist/controlplane.go

```diff
@@ -341,7 +378,15 @@ func (b *Botanist) DeployControlPlane(ctx context.Context) error {
 		}
 		return nil
 	})
-	return err
+	if err != nil {
+		return err
+	}
+
+	if b.isRestorePhase() {
+		return b.restoreExtensionObject(ctx, b.K8sSeedClient.Client(), cp, &cp.ObjectMeta, &cp.Status.DefaultStatus, extensionsv1alpha1.ControlPlaneResource, cp.Name, cp.Spec.DefaultSpec.GetExtensionPurpose())
```

Comment:
Why are we having 2 stage approach here. While creating extension object on new seed (i.e. at the time of restoration from shoot state) , we are first creating Object without status and then updating restore status in separate call. Why can't we have both spec and status for extension object in single call?

I would redefine the  `restoreExtensionObject` to simply update the extension object structure rather than actually updating it over k8s client. And move this `if` block inside the `CreateOrUpdate` call above. 

Also, the current way, in case of debugging operator won't able to figure out whether the extension object is from newly created object from the time of shoot creation or object creating at the time of restoration. There are such multiple open point we are leaving here. Like, everywhere else in code we have to take care of the case where `createOrUpdate` call succeeded but restoreStatus call failed might lead to introduction of code bugs in future.

Reply:
Is it possible to create an object with status?

Reply:
No, it's not. The `status` can only be set via the `/status` subresource, hence, two calls are needed.

> Also, the current way, in case of debugging operator won't able to figure out whether the extension object is from newly created object from the time of shoot creation or object creating at the time of restoration.

Yes, this boils down to my https://github.com/gardener/gardener/pull/2012#discussion_r407855927 where I was asking about a more explicit behaviour. The operations pov is a good argument as well.